erts/emulator/drivers/common/efile_drv.c

```diff
@@ -1261,26 +1261,61 @@ static void free_read_line(void *data)
     EF_FREE(d);
 }
 
+void read_file_zero_size(struct t_data* d);
+
+/* [ERL-327] Some special files like /proc/... have reported size 0 */
+void read_file_zero_size(struct t_data* d) {
+    size_t actual_size = 0;
+    while (1) {
```

Comment:
I personally prefer the `for (;;)` construct instead of `while (1)` since a compiler once warned me that `1` was a constant expression.